modules/standard/Regex.chpl

```diff
@@ -870,30 +882,34 @@ record regex {
    */
   iter split(text: exprType, maxsplit: int = 0)
   {
+    var localRegex = _regex;
+    var regexCopy:regex(exprType);
+    if home != here {
+      regexCopy = this;
+      localRegex = regexCopy._regex;
+    }
     var matches:_ddata(qio_regex_string_piece_t);
-    var ncaptures = qio_regex_get_ncaptures(_regex);
+    var ncaptures = qio_regex_get_ncaptures(localRegex);
     var nmatches = 1 + ncaptures;
     var pos:byteIndex;
     var endpos:byteIndex;
     var last:byteIndex;
+    var localText = text.localize();
 
-    on this.home {
-      matches = _ddata_allocate(qio_regex_string_piece_t, nmatches);
-    }
+    matches = _ddata_allocate(qio_regex_string_piece_t, nmatches);
```

Comment:
Shouldn't impact performance, but just to get rid of uses of _ddata, I think we should switch this to c_malloc/c_free (or c_calloc if we care about zero initing this)